src/shared/sam/debugger/awsSamDebugConfigurationValidator.ts

```diff
@@ -279,3 +283,50 @@ export class DefaultAwsSamDebugConfigurationValidator implements AwsSamDebugConf
         return { isValid: true }
     }
 }
+
+/**
+ * Resolves the `${workspaceFolder}` variable with the workspace folder.
+ * Unsure if any of the other variables are valuable for `aws-sam` configs at this time.
+ * @param folder
+ * @param config
+ * @returns resolved config
+ */
+export function resolveWorkspaceFolderVariable(
+    folder: vscode.WorkspaceFolder | undefined,
+    config: AwsSamDebuggerConfiguration
+): AwsSamDebuggerConfiguration {
+    return doTraverseAndReplace(config, folder?.uri.fsPath)
+}
+
+function doTraverseAndReplace(object: { [key: string]: any }, fspath: string | undefined): any {
+    const wsfRegex = /^(.*)(\$\{workspaceFolder\})(.*)$/g
+    if (!vscode.workspace.workspaceFolders && !fspath) {
+        throw new Error('No workspace folders available; cannot resolve workspaceFolder variable.')
+    }
+    const keys = Object.keys(object)
+    const final = JSON.parse(JSON.stringify(object))
+    for (const key of keys) {
+        const val = object[key]
+        if (typeof val === 'string') {
+            const result = wsfRegex.exec(val)
+            if (result) {
+                if (!fspath) {
+                    for (const wsf of vscode.workspace.workspaceFolders!) {
```

Comment:
I think the variables in configs are resolved by where the config is located. See here (bottom of the subsection): https://code.visualstudio.com/docs/editor/multi-root-workspaces#_debugging

So instead of iterating over all folders you should just find the folder where the config data was pulled from. Maybe I'm not understanding where the config information is coming from, but if my first statement is true then it would be easier to resolve the variable when the information is first extracted.

Reply:
We can't assume anything about the config file this is pulled from. Someone could write a config from scratch with `${workspaceFolder}` built in. I don't believe VS Code will give us the originating location of the launch config either (the command just returns a list of available launch configs).

Reply:
That's unfortunate, though this shouldn't be an issue except in very specific circumstances (user has multiple root folders with the same structure/file names).